### PR TITLE
[DEVOPS-1348] Test flyway using kubernetes

### PR DESCRIPTION
**Pull request by taksan**

### CODE REVIEW

File: core/src/test/java/objective/taskboard/domain/converter/IssueTeamServiceTest.java [hunk 1]
```diff
@@ -88,6 +88,7 @@ public void getMismatchingUsers_whenUsersAreNotValidAssignees_shouldReturnTheirN
         Set<String> mismatchingUsers = subject.getMismatchingUsers(i);
 
         assertEquals("fulano", StringUtils.join(mismatchingUsers));
+        assertEquals("break", "it");//TODO remove this; used to test pipepline
```
Comment (objective-solutions-user) [inline core/src/test/java/objective/taskboard/domain/converter/IssueTeamServiceTest.java]: MAJOR Swap these 2 arguments so they are in the correct order: expected value, actual value. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS3415)

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: Jenkinsfile [hunk 2]
```diff
@@ -1,131 +1,199 @@
...
-                        junit testResults: '**/target/failsafe-reports/*.xml', testDataPublishers: [[$class: 'AttachmentPublisher']], allowEmptyResults: true
+pipeline {
+    options {
+        disableConcurrentBuilds()
+    }
+    agent {
+        kubernetes {
+            label taskboardBuilderLabel
+            defaultContainer "taskboard-builder"
+            yaml """
```
Comment (herbert-venancio) [inline Jenkinsfile]: Documentation says that:
```
You can use readFile or readTrusted steps to load the yaml from a file.
```

File: Jenkinsfile [hunk 3]
```diff
@@ -1,131 +1,199 @@
...
+        - name: MYSQL_USER
+          value: taskboard 
+        - name: MYSQL_PASSWORD
+          value: taskboard 
+  - name: oracle11g
+    image: repo:5000/ng-oracle-11
+    securityContext:
+        privileged: true
+    tty: true
+    readinessProbe:
```
Comment (herbert-venancio) [inline Jenkinsfile]: It's missing some time-out parameters (Has any build failed to run oracle? We had some problems when directly running docker containers)

File: Jenkinsfile [hunk 4]
```diff
@@ -1,131 +1,199 @@
...
+      sizeLimit: "128Mi"
+  - name: repo-docker-creds
+    projected:
+      sources:
+      - secret:
+          name: repocred
+          items:
+            - key: .dockerconfigjson
+              path: .docker/config.json
+            """
```
Comment (herbert-venancio) [inline Jenkinsfile]: Unnecessary padding

File: Jenkinsfile [hunk 5]
```diff
@@ -1,131 +1,199 @@
...
+        }
+    }
+  
+    stages {
+        stage('Build') {
+            steps {
+                timeout(time: 60, unit: 'MINUTES') {
+                    checkout scm
+                    mvn "-U -Dmaven.test.failure.ignore=true clean install"
```
Comment (herbert-venancio) [inline Jenkinsfile]: `clean` is redundant

File: Jenkinsfile [hunk 6]
```diff
@@ -1,131 +1,199 @@
...
+                        }
+                    }
                 }
-            }, oracle: {
-                retry(2) {
-                    flyway.testOracleMigration()
+                stage('Flyway oracle') {
+                    steps {
+                        script {
+                            flywayModules.each {
```
Comment (herbert-venancio) [inline Jenkinsfile]: Missing `timeout(time: 1, unit: 'MINUTES')`

File: Jenkinsfile [hunk 7]
```diff
@@ -136,19 +204,63 @@ def isPostBuildBranch() {
...
+        "system", "oracle", 
+        location, 
+        schemaTable)
+}
+
+def flyway(module, database, jdbcUrl, user, password, location, schemaTable) {
+    vendor = jdbcUrl.split(':')[1]
+    sh """
+        cd $module
+        mvn org.flywaydb:flyway-maven-plugin:4.2.0:migrate \
```
Comment (herbert-venancio) [inline Jenkinsfile]: This plugin doesn't work correctly running in parallel in the same machine and folder. Previously there was a lock that didn't migrate to this version

File: Jenkinsfile [hunk 8]
```diff
@@ -1,131 +1,199 @@
...
+        port: 3306
+      initialDelaySeconds: 5
+      periodSeconds: 10
+    env:
+        - name: MYSQL_ROOT_PASSWORD
+          value: my-secret-pw
+        - name: MYSQL_DATABASE
+          value: taskboard 
+        - name: MYSQL_USER
+          value: taskboard
```
Comment (herbert-venancio) [inline Jenkinsfile]: Since we're using `root`, can we use only MYSQL_ROOT_PASSWORD environment variable?

File: Jenkinsfile [hunk 9]
```diff
@@ -136,19 +204,63 @@ def isPostBuildBranch() {
...
 }
 
-void clearDocker() {
-    def images = sh(script: 'sudo docker images -qf dangling=true', returnStdout: true).trim()
-    if(images)
-        sh(script: "sudo docker rmi $images", returnStatus: true)
+def flywayMariaDB(module, database, schemaTable, location) {
+    flyway(module, 
+        database, 
+        "jdbc:mysql://localhost:3306/taskboard?useUnicode=true&characterEncoding=UTF8",
```
Comment (herbert-venancio) [inline Jenkinsfile]: Maybe interpolate?
```groovy
"jdbc:mysql://localhost:3306/${database}?useUnicode=true&characterEncoding=UTF8"
```
Or... does it work without specifying a database on the connection string?

**Pull request merged**